pkgs/models/model.nix

```diff
@@ -0,0 +1,85 @@
+{ stdenvNoCC
+, fetchurl
+, makeWrapper
+
+, sticker
+
+# Word and tag embeddings as an attrset, containing the attributes:
+#
+# - embeds: Embedding file fixed-output derivation.
+# - filename: Embedding file name in the model's sticker.conf
+, wordEmbeds
+, tagEmbeds ? null
+
+# Short name of the model. E.g.: de-pos-ud.
+, modelName
+
+# Version of the model, typically a date. E.g.: 20190821
+, version
+
+, sha256
+}:
+
+rec {
+  model = stdenvNoCC.mkDerivation rec {
+    inherit version;
+
+    pname = "sticker-model-${modelName}";
+
+    src = fetchurl {
+      inherit sha256;
+
+      url = let
+        fullName = "${modelName}-${version}";
+        in "https://github.com/stickeritis/sticker-models/releases/download/${fullName}/${fullName}.tar.gz";
+    };
+
+    preConfigure = ''
+      substituteInPlace sticker.conf \
+        --replace "${wordEmbeds.filename}" \
+      "${wordEmbeds.embeds}"
```

Comment:
Not really sure what's happening here. You're replacing `${wordEmbeds.filename}` in `sticker.conf` with `${wordEmbeds.embeds}`. For the German UD POS model, `${wordEmbeds.filename}` is `de-structgram-20190426-opq.fifu`, `${wordEmbeds.embeds}` is the result of `fetchurl`, which would then be a URL?

Reply:
So, `sticker.conf` of the downloadable models only contains the embeddings filename such as `de-structgram-20190426-opq.fifu`. Of course, the embeddings could be placed in the same directory and the model would work. However, this would be a waste, because then every German model package would have its own copy of the same embeddings.

When the `fetchurl` expression is realized it results in a Nix store path such as `/nix/store/0zr42rpg75vxy3q3xbmnlpipbi8qg9bs-de-structgram-20190426-opq.fifu`. This path will **always** be the same, given the same URL and SHA256 hash (it's a so-called fixed-output derivation). Here I replace an embedding file name such as `de-structgram-20190426-opq.fifu` by its store path such as `/nix/store/0zr42rpg75vxy3q3xbmnlpipbi8qg9bs-de-structgram-20190426-opq.fifu`. Since it is a fixed-output derivation, the file name will be same for other model packages (e.g. German dependency parsing or German named entity tagging), so all German model packages will share the same embedding file.